book/glossary.md

```diff
@@ -0,0 +1,163 @@
+
+### Modularity
+
+Modular code is written in discrete, re-usable chunks. Similar code is kept close together, while pieces of code with different uses are stored separately. Breaking code down in this way makes code easier to work with, understand and review.
+
+### Readability
+
+Ability to gain an understanding of code within a reasonable amount of time.
+
+### Maintainability
+
+Ability to easily understand, modify and repair the code.
+
+## Tools
+
+### Open source
+
+A type of software license that allows users to do view the software's code. Open-source code is openly available to anyone. Using open-source software and publishing our code with open-source licenses ensures that our users can easily understand and reproduce our analysis.
+
+Open-source programming languages are free to use. We recommend using Python and R, which support the good practices outlined in this guidance.
+
+Proprietary software, including SPSS, SAS and Stata are closed source. These are expensive to use and do not support many of the good analysis practices outlined in this guidance. Not using open source analysis tools means that our users need to purchase software to reproduce our results.
+
+### Python and R
+
+Python and R are high-level programming languages that are commonly used in government analysis. They are both free to use and open source.
+
+### Git
+
+By far the most popular version control tool. Used to version changes to code for any programming language.
+
+### Version control platforms
+
+For example, GitHub, GitLab and BitBucket.
+
+These services use Git to manage remote repositories. Having a remote version of your repository allows multiple users to collaborate using Git. Each platform has free and paid options and allow users to open-source their work.
+
+Some platforms (e.g. GitLab) are open source and can be hosted internally by your organisation.
+
+### Continuous integration and continuous delivery (CI/CD)
+
+CI/CD tools help to automate parts of your code development workflow. This can include running automated tests against your code and deploying the latest version of your code into production. GitHub Actions is an example tool for running CI/CD workflows.
+
+Continuous integration describes regularly combining code changes from multiple contributors on a single software project. Integrating these changes regularly helps to check that independent changes work correctly together. Continous delivery describes automation of the software release process. Automating both of these using a CI/CD tool increases the efficiency of building software and getting it into production.
+
+### Virtual environment
+
+Virtual environments are tools for managing dependencies and isolating projects. A virtual environment is effectively a folder tree containing the programming language version and a list of dependencies. 
```

Comment:
Could add that this helps us to keep track of dependencies and avoid conflicts between different projects?